### PR TITLE
feat(Dropdown): add prop for clearing searchQuery data on item select or on enter

### PR DESCRIPTION
**Pull request by creedmangrum**
I wanted to add a prop to the dropdown where the search query could be cleared after selection on the multiple select, if desired. 

This maintains the current behavior of how everything works today, but if the prop is added then the search query will clear after item selection. 

I asked this question on issue #1954 but just figured it was easy enough to PR before waiting for an answer.

### CODE REVIEW

File: src/modules/Dropdown/Dropdown.d.ts [hunk 1]
```diff
@@ -216,6 +216,9 @@ export interface DropdownProps {
   /** A shorthand for a search input. */
   searchInput?: any;
 
+  /** Whether the searchQuery is cleared or not when selecting an item from search. */
+  searchQueryClearOnSelect?: boolean;
```
Comment (levithomason) [inline src/modules/Dropdown/Dropdown.d.ts]: Nit: Could we go with `clearSearchQueryOnSelect`?

**Review by levithomason — CHANGES_REQUESTED** [review]
I've requested a prop name change above.  We just need some tests for this prop:

```js
describe('clearSearchQueryOnSelect', () => {
  it('does not clear the query by default on item click', () => {})
  
  it('does not clear the query by default on enter', () => {})
  
  it('clears the query on item click', () => {})
  
  it('clears the query on enter', () => {})
})
```

File: src/modules/Dropdown/Dropdown.d.ts (hunk 1 above)
Comment (creedmangrum) [reply src/modules/Dropdown/Dropdown.d.ts]: not a nit at all, was hoping for some specific feedback on the naming anyway. i just put it near the search ones as it felt grouped, but really was unsure about naming convention. will switch to your suggestion.

**Comment by creedmangrum** [comment]
thanks for the feedback. will add the tests and prop name change.

**Comment by creedmangrum** [comment]
okay should be good to go!

File: test/specs/modules/Dropdown/Dropdown-test.js [hunk 2]
```diff
@@ -1659,6 +1659,80 @@ describe('Dropdown', () => {
...
+        .simulate('click')
+        .setState({ searchQuery })
+
+      wrapper
+        .find('DropdownItem')
+        .first()
+        .simulate('click', nativeEvent)
+
+      // cleared search query
+      wrapper.should.have.state('searchQuery', searchQuery)
```
Comment (levithomason) [inline test/specs/modules/Dropdown/Dropdown-test.js]: Comment needs updated, or removed :)

File: test/specs/modules/Dropdown/Dropdown-test.js [hunk 3]
```diff
@@ -1659,6 +1659,80 @@ describe('Dropdown', () => {
...
+
+      wrapper
+        .simulate('click')
+        .setState({ searchQuery })
+
+      domEvent.keyDown(document, { key: 'ArrowDown' })
+      domEvent.keyDown(document, { key: 'Enter' })
+
+      // cleared search query
+      wrapper.should.have.state('searchQuery', searchQuery)
```
Comment (levithomason) [inline test/specs/modules/Dropdown/Dropdown-test.js]: Comment out of date

**Review by levithomason — CHANGES_REQUESTED** [review]
There are a couple out of date comments that should be updated or removed.

We merge another PR which handles clearing the search query when there is only one option.  This PR has some conflicts that need to be resolved.

**Comment by creedmangrum** [comment]
Great taken care of. Sorry about the bad comments in the tests. Whoops.

**Comment by creedmangrum** [comment]
I think that the other PR that was merged probably solves for the use case, where the search query should be cleared if it is the available options === 1. I am likely going to close this, unless you feel that it still makes sense to do.

**Comment by levithomason** [comment]
Fewer options in the Dropdown, the happy I am 😄!  This component is quite complex and getting hairy to work on.

Assuming we're all good here, I'll take your suggestion and close this.  Let me know if we need to reopen it.

**Pull request closed without merging**